File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        queue = deque([root])

        while queue:
            Rnode = None
            qlen = len(queue)
            for i in range(qlen):
                curr = queue.popleft()
                if curr:
                    Rnode = curr
                    queue.append(curr.left)    
                    queue.append(curr.right)
            if Rnode:
                result.append(Rnode.val)
        return result

        

File: Data Structures & Algorithms/sum-root-to-leaf-numbers/submission-2.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int SumNumbers(TreeNode root) {
        int ans = 0;

        void dfs(TreeNode root, int curr){
            if(root == null) return;

            curr= curr*10 + root.val;

            if(root.left == null && root.right== null){
                ans+=curr;
                return;
            }
            dfs(root.left, curr);
            dfs(root.right, curr);
        }
        dfs(root, 0);
        return ans;
    }
}